src/main/java/com/me/artsafuanov/coursework3/service/impl/SocksServiceImpl.java

```
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class SocksServiceImpl implements SocksService {
+    private final Map<Socks, Integer> mapOfSocks = new HashMap<>();
+    private final FileSockServiceImpl fileSockService;
+
+    private final FileAuditServiceImpl fileAuditService;
+    private final AuditServiceImpl auditService;
+    public SocksServiceImpl(FileSockServiceImpl fileSockService, AuditServiceImpl auditService, FileAuditServiceImpl fileAuditService) {
+        this.fileSockService = fileSockService;
+        this.auditService = auditService;
+        this.fileAuditService = fileAuditService;
+    }
+    @PostConstruct
+    private void init() {
+        try {
+            fileSockService.readFromFile();
+            fileAuditService.readFromAuditFile();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    @Override
+    public void addSocks(SocksRequest socksRequest) {
+        validateRequest(socksRequest);
+        Socks socks = toSock(socksRequest);
+        if (mapOfSocks.containsKey(socks)) {
+            mapOfSocks.put(socks, mapOfSocks.get(socks) + socksRequest.getQuantity());
+        }else {
+            mapOfSocks.put(socks, socksRequest.getQuantity());
+            auditService.recordAddOperation(socks, socksRequest.getQuantity());
+            saveToFile();
+        }
+    }
+    @Override
+    public void outputSocks(SocksRequest socksRequest) {
+        decreaseSocks(socksRequest, true);
+    }
+
+    @Override
+    public void removeDefectSocks(SocksRequest socksRequest) {
+        decreaseSocks(socksRequest, false);
+    }
+    private void decreaseSocks (SocksRequest socksRequest, boolean isOutput) {
+        validateRequest(socksRequest);
+        Socks socks = toSock(socksRequest);
+        int socksQuantity = mapOfSocks.getOrDefault(socks, 0);
+        if (socksQuantity >= socksRequest.getQuantity()) {
```

лесенка получается, после первого if сразу второй идет. 
fail fast или early return  
пишем один if проверяющий и выкидывающий ошибку. Тогда else не нужно будет писать вообще.

Это сильно разгружает подобный код